src/scales/scale.ts

```diff
@@ -122,19 +118,23 @@ module Plottable {
       }
     }
 
-    protected _getDomain() {
-      return this._d3Scale.domain();
+    protected _getDomain(): D[] {
+      throw new Error("subclasses should override _getDomain");
     }
 
     protected _setDomain(values: D[]) {
       if (!this._domainModificationInProgress) {
         this._domainModificationInProgress = true;
-        this._d3Scale.domain(values);
+        this._setActualDomain(values);
         this._dispatchUpdate();
         this._domainModificationInProgress = false;
       }
     }
 
+    protected _setActualDomain(values: D[]) {
```

Comment:
This is specifically to set the domain of usually the backing D3 scale.  I was completely unsure of the name as it's basically `_setDomain()` of the backing scale but I wasn't sure of how to go about it.  I mainly didn't want to rewrite the previous implementation of `setDomain()` over and over as that can lead to inconsistencies.